### ISSUE TITLE
Boats stop randomly

### ISSUE THREAD

**Issue by HoodBlah** [MuradAkh/LittleLogistics#194]
Forge 1.18.2
I have terraforged installed so there are large rivers going all over the place.
I set up a loop for my electric tugboats, a starting dock, checkpoint dock, and end dock.
They carried a chunk loader, a seat, 3 chests, and 2 fishers.
It completed half of a loop, but 3-4 nodes before entering the checkpoint dock a second time it stopped working.
The battery was nearly full, the tag/map thing has about 130 nodes, but it stopped at the 90th.
After making a new steam boat with the same carriers and a smaller map with of about 30 nodes, I continued.
It ended up stopping again, 3 nodes before the starting dock again.
I don't know If this is a user error since there were a lot of nodes but I made sure they were in the correct order and that there were no overlaps.

**Comment by MuradAkh** [comment]
It sounds like a bug, probably doesn't flag the node as visited for some reason. Nevertheless, you generally don't need that many nodes as the tugs are pretty good at navigating the rivers. 

Does it get stuck right by the node and just wait randomly? Does reinserting the route help?

Unfortunately these kinds of issues are very hard to debug, any chance you could share a world download?

Does it seem to happen randomly or does it fail at a specific spot? If it's a specific spot then few screenshots would be helpful too

**Comment by HoodBlah** [comment]
It does get stuck right by the node. Ive tried reinserting the route, but it remains stuck and the progress gets reset. 
I put a lot of nodes in because the docks are over 300 blocks away from each other, and because the river turns a lot and has edges that go over the shores. 
I tried letting the tug navigate by itself, but it would get stuck on the sides of the river (and go inside the sand). 
It seems to only fail a couple of nodes before the start of the dock.

Generally, I placed the nodes about 5-10 blocks apart.

I am playing with some heavy mods so the texture might be different, however, the terraforged world that I have already rendered should still be there as is.

I dont know of the Xaero's minimap waypoints will show for you; but there is a village far north and down stream, along the way you should find a dock with the boat, going down stream and to the left there is another checkpoint dock on the sand, then finally one by the village (I had the boat go a little past it so it can make a u-turn).

[wOrld.zip](https://github.com/MuradAkh/LittleLogistics/files/8951663/wOrld.zip)

P.S. I also noticed that when going through waterfalls, the tug will go a bit to the side to line up with the next node. As some of the nodes on the return trip are close to the edge of the water fall, I feel that could be one of the reasons for this bug.

**Comment by HoodBlah** [comment]
Another AI issue, this time the boat just started going straight into the shore.
I tried doing the circuit again with only 30 nodes this time.
It was going smoothly, except whenever it went near a waterfall it started going off to the side trying to avoid it, ultimately beaching itself.
I used a lead to pull it back into the water and fixed the beached area so it doesn't happen again.
This happened a couple times until it got to the fork in the river.
It just started going straight and into the other end of the fork, beaching itself. I tried using leads and turning off the engine but it kept going even with the engine off!

https://photos.app.goo.gl/zR45g29hPjMGsB9s7
The video was too big to put here, so this is a google drive link to it.

https://drive.google.com/file/d/1p0Nnf-JgBit-fqzBkd2IDVLafUJS_EYY/view?usp=sharing
The world is also to big apparently?
As soon as I finished the video, I left everything as is. So it should have the route and boat in it.
You'll find it at the fork down the river.

**Comment by MuradAkh** [comment]
Looks like tug AI breaks for you for some reason. Do you have any optimisation mods installed that could be affecting mob AI?

Do you have optifine installed?

**Comment by HoodBlah** [comment]
I have Optifine, entityculling, and AI improvements. These are the ones I have that affect entities.

**Comment by MuradAkh** [comment]
I wouldn't be surprised if that was the root cause. Could you please remove these mods and confirm whether it happens again. If that's the case please report it to the respective mod. 

Please let me know if you find the mod that caused this. My guess it's not EntityCulling. 

As an alternative to optifine, I recommend using Rubidium. 

Also, if you insist on using optifine, you should follow the instructions in the pinned issue to make sure the chunkloader works.

**Comment by HoodBlah** [comment]
I replaced Optifine with rubidium and it is working now. The other mods don't really seem to change anything. There is still the issue of the boat sometimes going to the side to avoid waterfalls, and then stopping for a moment after going down one, but that isn't really an issue.

**Comment by HoodBlah** [comment]
Nvm, it started happening again. After a bit of experimenting I can say that the problem seems to happen after using a lead on the boats. Maybe something about forcing it to move breaks the AI?

**Comment by MuradAkh** [comment]
You've mentioned earlier that it happened independently though when you just set off the tug on a path. Could you confirm if that still happens? I could certainly investigate the lead issue

**Comment by HoodBlah** [comment]
Yes, the previous times it happened I used leads to pull the tugs out of the shore. The bug only occurred shortly after using the leads, it did not show up immediately.

As for it happening independently, that was after the initial bug appearance. As the bug forms in the tug by either moving in a direction or staying still.

**Comment by DogeOrTrain** [comment]
I have the same issue, i go out of chunks with my tug boat and then when i come back it stands still for no reason. The only way I can make it move again is by either going out of chunks then coming back, or destroying the boat.

I also have Optifine, EntityCulling and AI Improvements, I'll think of removing them first.

**Comment by HoodBlah** [comment]
It’s probably Optifine, the others don’t seem to do anything to the boats.

**Comment by Ghostipedia** [comment]
I have had a similar problem with boats stopping/halting for uncertain reasons
- I play in a server environment. So client mods like Optifine & EntityCulling Don't Exist
- I do not have AI Improvements 
- I have exclusively used Electric Tugs Prior to this problem
- The Electric Tugs will not even move to their first node from a station, they will sit by the charging station/dock idle, but still consume power
- Prior to this issue occurring today I only ever had one Tug in the water that only had one station : For Context It was 
1x Electric Tug 2x Fishing Barge 1x Chunk Loader 1x Seat
- I have tested 
Making a new tug, giving it the route controller 
New Tug, new route entirely
New Tug, New Dock, New Route, lots of nodes
New Tug, New Dock, New Route, few nodes

Things I did today, prior to noticing things breaking which MAY have been part of the problem
- I added another electric tug to the waters, and it shares a dock with the original tug (it's an unloading bay) 
- They were working for some time but the only "ADMIN" class action I've taken on my server in a long while was removing my end and nether dimensions
- For the first time I've had a graphical bug, reason I mention it is because I am unsure if it was 100 percent graphical, as problems arised after it, basically looked as if one of the 'boat chain's was attached to me, from very far away.
- I updated some mods on the 22nd? Unsure if this could be the issue, as the problem started on the 23rd, and was running perfectly fine the entire day of the 22nd with the updated mods, though if it is part of the problem, do know what mod jars have been recently modified and can provide that list as well

I'm running LL in a medium/large pack (about 180 mods)  and it's been working consistently fine for weeks IRL time so this problem cropping up has been a recent issue.
The Peculiar issue here is it ONLY applies to energy tugs, steam tugs work 100% as intended, they path, stop, restock and so on.
My concern is if this issue is something that actively ruins tugs, I've held off keeping steam tugs in the water to prevent any risks of the AI breaking.
If any info like logs, data files from the world, or screenshots or videos needed, I will provide them as well.
Forge Version : 40.1.51
LL Version : v1.2.5.

**Comment by Ghostipedia** [comment]
Here's a video of what It's doing, although with some surprising developments 
https://www.youtube.com/watch?v=C9qU7ky2hUg

**Comment by MuradAkh** [comment]
Interesting, this looks like a seperate issue to me

**Comment by HoodBlah** [comment]
Is there any update on what the issue was?
